packages/medusa/src/services/discount.ts

```diff
@@ -483,9 +442,8 @@ class DiscountService extends BaseService {
         )
         toCreate.ends_at = lastValidDate
       }
-      const created = await discountRepo.create(toCreate)
-      const result = await discountRepo.save(created)
-      return result
+      const created: Discount = await discountRepo.create(toCreate)
```

Comment:
todo,very nit: Remove await
```suggestion
      const created: Discount = discountRepo.create(toCreate)
```